libp2p/transports/tcptransport.nim

```diff
@@ -26,10 +26,8 @@ const
 type
   TcpTransport* = ref object of Transport
     server*: StreamServer
-    clients: seq[StreamTransport]
+    clients: array[bool, seq[StreamTransport]]
```

Comment:
wait what? `array[bool, seq[StreamTransport]]` should be a compiler error

Reply:
Nope, works as expected. Handy trick for when you have only two types of objects, like `incoming == true/false`.

Reply:
This is very confusing, why not use a enum type instead?
Bool flags are bad in general + array is more convoluted. I get the handy trick part but its very unclear.
```
MyType = object
  kind side
  of incoming:
    inClients: seq[...]
  of outgoing:
    outClients: seq[...]
```

Reply:
Altho applied to code array looks better, still a hack emulating rust `match` tho... fine for me tho in the end! 😄

Reply:
could also use the `Direction` enum, ie `array[Direction, ...]` now that it's binary